daos/TuitDao.ts

```diff
@@ -7,45 +7,122 @@ import Tuit from "../models/tuits/Tuit";
+            .sort({ 'postedOn': -1 })
             .populate("postedBy")
             .exec();
+
+    /**
+      * Uses TuitModel to retrieve single tuit document from tuits collection
+      * @param {string} tid Tuit's primary key
+      * @returns Promise To be notified when tuit is retrieved from the database
+      */
     findTuitById = async (tid: string): Promise<any> =>
         TuitModel.findById(tid)
             .populate("postedBy")
             .exec();
+
+    /**
+      * Inserts tuit instance into the database with user id
+      * @param {string} uid User's primary key
+      * @param {Tuit} tuit Instance to be inserted into the database
+      * @returns Promise To be notified when tuit is inserted into the database
+      */
     createTuitByUser = async (uid: string, tuit: Tuit): Promise<Tuit> =>
-        TuitModel.create({...tuit, postedBy: uid});
+        TuitModel.create({ ...tuit, postedBy: uid });
+
+    /**
+     * Updates tuit with new values in database
+     * @param {string} tid Tuit's primary key
+     * @param {Tuit} tuit Tuit object containing properties and their new values
+     * @returns Promise To be notified when tuit is updated in the database
+     */
     updateTuit = async (tid: string, tuit: Tuit): Promise<any> =>
         TuitModel.updateOne(
-            {_id: tid},
-            {$set: tuit});
+            { _id: tid },
+            { $set: tuit });
+
+    /**
+     * Updates likes of the tuit with new values in database
+     * @param {string} tid Tuit's primary key
+     * @param {any} newStats Tuit status containing likes number
+     * @returns Promise To be notified when tuit like status is updated in the database
+     */
     updateLikes = async (tid: string, newStats: any): Promise<any> =>
         TuitModel.updateOne(
-            {_id: tid},
-            {$set: {stats: newStats}}
+            { _id: tid },
+            { $set: { stats: newStats } }
         );
+
+    /**
+     * Removes a tuit from the database.
+     * @param {string} tid Tuit's primary key
+     * @returns Promise To be notified when the tuit is removed from the database
+     */
     deleteTuit = async (tid: string): Promise<any> =>
-        TuitModel.deleteOne({_id: tid});
+        TuitModel.deleteOne({ _id: tid });
+
+    /**
+      * Removes all tuits from the database.
+      * @returns Promise To be notified when all tuits are removed from the database
+      */
     deleteAllTuits = async (): Promise<any> =>
         TuitModel.deleteMany({});
+
+    /**
+      * Inserts tuit instance with images into the database with user id
+      * @param {string} uid User's primary key
+      * @param {Tuit} tuit Instance to be inserted into the database
+      * @param {Array<String>} image image array posted with the tuit
+      * @returns Promise To be notified when tuit is inserted into the database
+      */
+    createTuitByUserWithImage = async (uid: string, tuit: Tuit, image: Array<String>): Promise<Tuit> =>
```

Comment:
Actually, now think about it, this would be different from before. The client will send images/video as files through an HTTP request, instead of ```image: Array<String>```.

Can you close this PR and assign the task back to me?